specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-04-01-preview/workspaceRP.json

```diff
@@ -870,38 +855,34 @@
           {
             "$ref": "#/parameters/WorkspaceNameParameter"
           },
+          {
+            "$ref": "#/parameters/ConnectionName"
+          },
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
           {
             "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
           },
           {
             "in": "body",
             "name": "body",
-            "description": "The parameter of diagnosing workspace health",
+            "description": "deployment object",
+            "required": true,
             "schema": {
-              "$ref": "#/definitions/DiagnoseWorkspaceParameters"
+              "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
             }
           }
         ],
         "responses": {
           "200": {
             "description": "The request was successful; the request was well-formed and received properly.",
             "schema": {
-              "$ref": "#/definitions/DiagnoseResponseResult"
+              "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
             }
           },
           "202": {
```

Comment:
`put` should use `200` and `201` only

Reply:
We would like to request an exception on this, this API is a migration of an existing API so we want to keep the behavior the same, we have already made sure we return 200/201 for our newly added APIs.

I will put the lint error in suppression to unblock the review.

Reply:
could you provide the swagger link for the existing API?

Reply:
https://github.com/Azure/azure-rest-api-specs/blob/8d1d3d29cdbe03f95a9b4c0ec6bb2221f1c44bde/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json#L1267

this is the existing API, we want to migrate the user to this new one and keep the schema the same